table.cc

```diff
@@ -723,8 +723,7 @@ table::update_cache(lw_shared_ptr<memtable> m, sstables::shared_sstable sst) {
     if (_config.enable_cache) {
         return _cache.update(adder, *m);
     } else {
-        adder();
-        return m->clear_gently();
+        return _cache.invalidate(adder).then([m] { return m->clear_gently(); });
```

Comment:
Why do we need the clear_gently?

Reply:
I'm not sure. I just left what was there before. @tgrabiec probably knows best.

Reply:
Ah, of course you need it, to remove stuff from the memtable. I didn't look at the code thoroughly and thought it was about the cache.